### PR TITLE
Switch to Jctools NBHML and FieldUpdater

### PR DESCRIPTION
**Pull request by tjake**
EXISTING:
```
PQ@384 build 27.32s,
PQ encoded 100000 vectors [41.15 MB] in 3.26s,
Build M=8 ef=60 in 73.49s with 0.60 short edges
  Query PQ=true top 101/1 recall 0.7980 in 4.16s after 19444540 nodes visited
  Query PQ=true top 101/2 recall 0.9303 in 5.40s after 32572630 nodes visited
Build M=8 ef=80 in 13.96s with 0.63 short edges
  Query PQ=true top 101/1 recall 0.8074 in 3.50s after 20492178 nodes visited
  Query PQ=true top 101/2 recall 0.9362 in 5.20s after 34097234 nodes visited
Build M=8 ef=100 in 15.27s with 0.66 short edges
  Query PQ=true top 101/1 recall 0.8090 in 3.62s after 20328850 nodes visited
  Query PQ=true top 101/2 recall 0.9410 in 5.41s after 34196102 nodes visited
```

WITH NBHM:
```
PQ@384 build 27.03s,
PQ encoded 100000 vectors [41.15 MB] in 3.14s,
Build M=8 ef=60 in 12.80s with 0.59 short edges
  Query PQ=true top 101/1 recall 0.8038 in 3.94s after 18803416 nodes visited
  Query PQ=true top 101/2 recall 0.9323 in 4.97s after 31918654 nodes visited
Build M=8 ef=80 in 16.09s with 0.63 short edges
  Query PQ=true top 101/1 recall 0.8085 in 3.42s after 19716632 nodes visited
  Query PQ=true top 101/2 recall 0.9388 in 5.56s after 33320638 nodes visited
Build M=8 ef=100 in 18.57s with 0.66 short edges
  Query PQ=true top 101/1 recall 0.8117 in 3.45s after 20690602 nodes visited
  Query PQ=true top 101/2 recall 0.9416 in 5.29s after 34765870 nodes visited
```

WITH MBHML:
```
PQ@384 build 26.53s,
PQ encoded 100000 vectors [41.15 MB] in 3.14s,
Build M=8 ef=60 in 16.43s with 0.60 short edges
  Query PQ=true top 101/1 recall 0.8053 in 3.97s after 18712402 nodes visited
  Query PQ=true top 101/2 recall 0.9355 in 5.24s after 31836994 nodes visited
Build M=8 ef=80 in 12.77s with 0.63 short edges
  Query PQ=true top 101/1 recall 0.8082 in 3.41s after 19770528 nodes visited
  Query PQ=true top 101/2 recall 0.9402 in 5.11s after 33250648 nodes visited
Build M=8 ef=100 in 14.83s with 0.66 short edges
  Query PQ=true top 101/1 recall 0.8103 in 3.36s after 20285888 nodes visited
  Query PQ=true top 101/2 recall 0.9419 in 5.24s after 34188580 nodes visited
```

### CODE REVIEW

File: jvector-base/src/main/java/io/github/jbellis/jvector/graph/OnHeapGraphIndex.java [hunk 1]
```diff
@@ -91,7 +96,7 @@ public void addNode(int node) {
 
   /** must be called after addNode once neighbors are linked in all levels. */
   void markComplete(int node) {
-    entryPoint.accumulateAndGet(
+    entryPointUpdater.accumulateAndGet(this,
```
Comment (jbellis) [inline jvector-base/src/main/java/io/github/jbellis/jvector/graph/OnHeapGraphIndex.java:99]: I believe this is equivalent to the old code, the reason to use an Updater is when you have multiple long fields and you want to save the memory overhead of wrapping each in an Atomic

Comment (tjake) [reply jvector-base/src/main/java/io/github/jbellis/jvector/graph/OnHeapGraphIndex.java:99]: It failed for me which is why I changed it

Comment (jkni) [reply jvector-base/src/main/java/io/github/jbellis/jvector/graph/OnHeapGraphIndex.java:99]: @tjake was your failure of the form "Entry node was incompletely added!"?

Comment (tjake) [reply jvector-base/src/main/java/io/github/jbellis/jvector/graph/OnHeapGraphIndex.java:99]: I will try and undo this in #113 closing this one

**Pull request closed without merging**